ecl/eclcc/WORKUNITS.rst

```diff
@@ -0,0 +1,862 @@
+=======================
+Understanding workunits
+=======================
+
+Introduction
+************
+
+A workunit contains all the information that the system requires about a query - including what parameters it takes,
+how to execute it, and how to format the results.  Understanding the contents of a workunit is a key step to
+understanding how the HPCC system fits together.  This document begins with an overview of the different elements
+in a workunit.  That is followed by a walk-through executing a simple query, with a more detailed description of
+some of the workunit components, to show how they all tie together.
+
+Before looking at the contents of the workunit it is important to understand one of the design goals behind the HPCC
+system.  The ECL system logically splits that code that is needed to execute a query in two.  One the one hand there
```

Comment:
"The ECL system..." this is the first time you mention ECL, a brief lead into this would be good or remove the ref altogether for now.
"splits **the** code that..."
"**On** the one and..."